test/e2e/generate/generate_test.go

```diff
@@ -0,0 +1,288 @@
+package generate
+
+import (
+	"errors"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"os"
+	"testing"
+	"time"
+)
+
+var (
+	// Cluster Polict GVR
+	clPolGVR = GetGVR("kyverno.io", "v1", "clusterpolicies")
+	// Namespace GVR
+	nsGVR = GetGVR("", "v1", "namespaces")
+	// ClusterRole GVR
+	crGVR = GetGVR("rbac.authorization.k8s.io", "v1", "clusterroles")
+	// ClusterRoleBinding GVR
+	crbGVR = GetGVR("rbac.authorization.k8s.io", "v1", "clusterrolebindings")
+	// Role GVR
+	rGVR = GetGVR("rbac.authorization.k8s.io", "v1", "roles")
+	// RoleBinding GVR
+	rbGVR = GetGVR("rbac.authorization.k8s.io", "v1", "rolebindings")
+
+	// ClusterPolicy Namespace
+	clPolNS = ""
+	// Namespace Name
+	// Hardcoded in YAML Definition
+	nspace = "test"
+)
+
+func Test_ClusterRole_ClusterRoleBinding_Sets(t *testing.T) {
+	RegisterTestingT(t)
+	if os.Getenv("E2E") == "" {
+		t.Skip("Skipping E2E Test")
+	}
+	// Generate E2E Client ==================
+	e2eClient, err := NewE2EClient()
+	Expect(err).To(BeNil())
+	// ======================================
+
+	// ====== Range Over ClusterRoleTests ==================
+	for _, tests := range ClusterRoleTests {
+		By(fmt.Sprintf("Test to generate ClusterRole and ClusterRoleBinding : %s", tests.TestName))
+		By(fmt.Sprintf("synchronize = %v\t clone = %v", tests.Sync, tests.Clone))
+
+		// ======= CleanUp Resources =====
+		By(fmt.Sprintf("Cleaning Cluster Policies from Namespace : %s", clPolNS))
+		e2eClient.CleanClusterPolicies(clPolGVR, clPolNS)
+		// Clear Namespace
+		By(fmt.Sprintf("Deleting Namespace : %s\n", tests.ResourceNamespace))
+		e2eClient.DeleteClusteredResource(nsGVR, tests.ResourceNamespace)
+
+		// Wait Till Deletion of Namespace
+		GetWithRetry(time.Duration(1), 15, func() error {
+			_, err := e2eClient.GetClusteredResource(nsGVR, tests.ResourceNamespace)
+			if err != nil {
+				return nil
+			}
+			return errors.New("Deleting Namespace")
+		})
+
+		// If Clone is true Clear Source Resource and Recreate
+		if tests.Clone {
+			By(fmt.Sprintf("Clone = true, Deleting Source ClusterRole and ClusterRoleBinding from Clone Namespace : %s\n", tests.CloneNamespace))
+			// Delete ClusterRole to be cloned
+			e2eClient.DeleteNamespacedResource(crGVR, tests.CloneNamespace, tests.ClonerClusterRoleName)
+			// Delete ClusterRoleBinding to be cloned
+			e2eClient.DeleteNamespacedResource(crbGVR, tests.CloneNamespace, tests.ClonerClusterRoleBindingName)
+		}
+
+		// Wait to Delete Resources
+		// time.Sleep(10 * time.Second)
```

Comment:
Do we need to wait here?

Reply:
I am going to delete this as I have added Retry Mechanism